speech/unit_tests/test_client.py

```diff
@@ -500,16 +500,16 @@ def test_stream_recognize_interim_results(self):
 
         alternatives = [{
             'transcript': 'testing streaming 1 2 3',
-            'confidence': 0.9224355,
+            'confidence': 0.125,
         }, {
             'transcript': 'testing streaming 4 5 6',
-            'confidence': 0.0123456,
+            'confidence': 0.125,
```

Comment:
For all your other random values, I generated 20 more fractions that can be represented exactly in both `float` and `double` values

```
   0.1796875  # 23 / 128
   0.0546875  # 7 / 128
    0.390625  # 25 / 64
         0.5  # 1 / 2
0.5888671875  # 603 / 1024
         0.5  # 1 / 2
       0.875  # 7 / 8
  0.19140625  # 49 / 256
     0.84375  # 27 / 32
         0.5  # 1 / 2
      0.4375  # 7 / 16
         0.5  # 1 / 2
        0.75  # 3 / 4
        0.75  # 3 / 4
      0.4375  # 7 / 16
 0.314453125  # 161 / 512
    0.203125  # 13 / 64
0.6396484375  # 655 / 1024
 0.740234375  # 379 / 512
         0.5  # 1 / 2
```